### ISSUE TITLE
 Train Test Split issue?

### ISSUE THREAD

**Issue by pit9921** [pycaret/pycaret#1409]
I am not sure if i understand the functionality of pycaret well.
They are two dataframes:
- train
- test

`reg = setup(data = train, target='Output')`
As default it takes train_size = 0.7.

But, **i do not need any split (train / test) in the setup**, because these are defined before the setup is being done.
How can i prevent the "train_size " in the set up at all?
I have tried to set it to 1 (all data ("train")) should be used by model building, but the maximum is 0.9999.

What does this split (train_size) in case that by model building the only train data are given?

Does exist any explanation for that?

**Comment by IncubatorShokuhou** [comment]
Actually, the `seen` dataset is splitted into two datasets: `training set` and `validation set`. Only when you get good results on both the `training set` and the `validation set` can you apply the model to the `test` set. To use all the dataset, you need to use `finalize_model` to finalize your  model with super parameters determined with all your `seen` dataset.

**Comment by tvdboom** [comment]
I believe what you are looking for is the `test_data` parameter in setup. Since you say that you already have a train and test set defined before the setup, you can add your test dataset to `test_data` as a pd.DataFrame with the same number of columns as `data`. If you do, the `train_size` parameter is ignored, the dataframe passed to `data` is used as training set and the dataframe passed to `test_data` is used as test set.

**Comment by pit9921** [comment]
Thank you tvdboom. The parameter "test_data" is what i was looking for 👍
